### PR TITLE
Refactoring CloneGitRepoAPI.spec.ts to work with DevWorkspace generator tool

### LINKED ISSUE #22223: Refactoring CloneGitRepoAPI.spec.ts to work with DevWorkspace generator tool
After new feature #22212 added to the che test automation framework API tests can be improved to use it. In current scope there is one spec:  `CloneGitRepoAPI.spec.ts`  should be updated. 

Related comment: https://github.com/eclipse/che/issues/22221#issuecomment-1554205509:
> @nallikaea: taking into account, that `DevfileAcceptanceTestAPI.spec.ts` will "_Verify that the project has been cloned successfully_", what do you think about renaming `CloneGitRepoAPI.spec.ts` test to `EmptyWorkspaceAPI.spec.ts`, to avoid confusion in future?_

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?
- CloneGitRepoAPI.spec.ts renamed to EmptyWorkspaceAPI.spec.ts
- Test refactored to use [DevWorkspaceGenetator tool](https://github.com/eclipse/che/issues/22212)
- API Constants updated
- DevWorkspaceConfigurationHelper upadted to use test constants

### Screenshot/screencast of this PR

NA

### What issues does this PR fix or reference?

#22223 

### How to test this PR?

export USERSTORY=EmptyWorkspaceAPI && npm run driver-less-test

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nallikaea** [comment]
```
/usr/bin/npm run driver-less-test

> @eclipse-che/che-e2e@7.68.0-SNAPSHOT driver-less-test
> export TS_USE_WEB_DRIVER_FOR_TEST=false && npm run test


> @eclipse-che/che-e2e@7.68.0-SNAPSHOT test
> ./configs/sh-scripts/initDefaultValues.sh npm run lint && npm run tsc && export USERSTORY=$USERSTORY && mocha --config dist/configs/mocharc.js

Initialized default values

TS_SELENIUM_VALUE_TLS_SUPPORT =       
TS_SELENIUM_VALUE_OPENSHIFT_OAUTH =   true
TS_OCP_LOGIN_PAGE_PROVIDER_TITLE =    htpasswd
E2E_OCP_CLUSTER_VERSION =             4.x
TS_SELENIUM_LOG_LEVEL =               TRACE
TS_SELENIUM_OCP_USERNAME =            admin
TS_SELENIUM_W3C_CHROME_OPTION =       true
NODE_TLS_REJECT_UNAUTHORIZED =        0
TS_SELENIUM_EDITOR =                  che-code

> @eclipse-che/che-e2e@7.68.0-SNAPSHOT tsc
> rm -rf ./dist && ./configs/sh-scripts/generateIndex.sh && tsc -p .

Generating index.ts file...

################## Launch Information ##################

      TS_SELENIUM_BASE_URL: https://devspaces.apps.ocp412-mdolhal.crw-qe.com
      TS_SELENIUM_HEADLESS: false
      TS_SELENIUM_OCP_USERNAME: admin
      TS_SELENIUM_EDITOR:   che-code

      TS_SELENIUM_HAPPY_PATH_WORKSPACE_NAME: petclinic-dev-environment
      TS_SELENIUM_DELAY_BETWEEN_SCREENSHOTS: 1000
      TS_SELENIUM_REPORT_FOLDER: ./report
      TS_SELENIUM_EXECUTION_SCREENCAST: false
      DELETE_SCREENCAST_IF_TEST_PASS: true
      TS_SELENIUM_REMOTE_DRIVER_URL: 
      DELETE_WORKSPACE_ON_FAILED_TEST: false
      TS_SELENIUM_LOG_LEVEL: TRACE
      TS_SELENIUM_LAUNCH_FULLSCREEN: true

      TS_COMMON_DASHBOARD_WAIT_TIMEOUT: 5000
      TS_SELENIUM_START_WORKSPACE_TIMEOUT: 360000
      TS_WAIT_LOADER_PRESENCE_TIMEOUT: 60000

      TS_SAMPLE_LIST: Node.js MongoDB,Node.js Express

      MOCHA_DRIRECTORY is not set
      USERSTORY is not set

      to output timeout variables, set TS_SELENIUM_PRINT_TIMEOUT_VARIABLES to true
 ######################################################## 


            ‣ DriverHelper.getDriver
  Empty workspace API test
          ▼ DevWorkspaceConfigurationHelper.generateDevfileContext
No plug-in registry url. Setting to https://eclipse-che.github.io/che-plugin-registry/main/v3
(node:207829) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
DevWorkspace che-code-empty-715 was generated.
          ▼ KubernetesCommandLineToolsExecutor.loginToOcp - oc: Login to the "OC" client.
          ▼ KubernetesCommandLineToolsExecutor.getServerUrl - oc: Get server api url.
oc whoami && oc whoami --show-server=true
error: You must be logged in to the server (Unauthorized)
          ▼ KubernetesCommandLineToolsExecutor.loginToOcp - oc: Login https://api.ocp412-mdolhal.crw-qe.com:6443, admin
Login successful.

You have access to 70 projects, the list has been suppressed. You can list all projects with 'oc projects'

Using project "default".
          ▼ DevWorkspaceConfigurationHelper.getDevWorkspaceConfigurationYamlAsString
          ▼ KubernetesCommandLineToolsExecutor.applyYamlConfigurationAsStringOutput - oc:
cat <<EOF | oc apply -n admin-devspaces -f - 
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspaceTemplate
metadata:
  name: che-code-empty-715
spec:
  commands:
    - id: init-container-command
      apply:
        component: che-code-injector
    - id: init-che-code-command
      exec:
        component: che-code-runtime-description
        commandLine: nohup /checode/entrypoint-volume.sh > /checode/entrypoint-logs.txt
          2>&1 &
  events:
    preStart:
      - init-container-command
    postStart:
      - init-che-code-command
  components:
    - name: che-code-runtime-description
      container:
        image: quay.io/devfile/universal-developer-image@sha256:98264e81d11bcfd74d46fd2037f5d26ee37794bb57b033328f35d838642e297f
        volumeMounts:
          - name: checode
            path: /checode
        memoryLimit: 1024Mi
        memoryRequest: 256Mi
        cpuLimit: 500m
        cpuRequest: 30m
        endpoints:
          - name: che-code
            attributes:
              type: main
              cookiesAuthEnabled: true
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: code-redirect-1
            attributes:
              discoverable: false
              urlRewriteSupported: false
            targetPort: 13131
            exposure: public
            protocol: http
          - name: code-redirect-2
            attributes:
              discoverable: false
              urlRewriteSupported: false
            targetPort: 13132
            exposure: public
            protocol: http
          - name: code-redirect-3
            attributes:
              discoverable: false
              urlRewriteSupported: false
            targetPort: 13133
            exposure: public
            protocol: http
      attributes:
        app.kubernetes.io/component: che-code-runtime
        app.kubernetes.io/part-of: che-code.eclipse.org
        controller.devfile.io/container-contribution: true
    - name: checode
      volume: {}
    - name: che-code-injector
      container:
        image: quay.io/che-incubator/che-code@sha256:235f2028e1869ac4630a61adbc5cd33e75584275f1789eb5cd5ad36ff6713a6a
        command:
          - /entrypoint-init-container.sh
        volumeMounts:
          - name: checode
            path: /checode
        memoryLimit: 256Mi
        memoryRequest: 32Mi
        cpuLimit: 500m
        cpuRequest: 30m
---
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspace
metadata:
  name: empty-715
  annotations:
    che.eclipse.org/devfile: |
      schemaVersion: 2.2.0
      metadata:
        name: empty-715
spec:
  started: true
  routingClass: che
  template: {}
  contributions:
    - name: editor
      kubernetes:
        name: che-code-empty-715

EOF
devworkspacetemplate.workspace.devfile.io/che-code-empty-715 created
devworkspace.workspace.devfile.io/empty-715 created
          ▼ KubernetesCommandLineToolsExecutor.waitDevWorkspace - oc: Wait till workspace ready.
oc wait -n admin-devspaces --for=condition=Ready dw empty-715 --timeout=360s
devworkspace.workspace.devfile.io/empty-715 condition met
          ▼ KubernetesCommandLineToolsExecutor.getWorkspacePodName - oc: Get workspace pod name.
oc get pod -l controller.devfile.io/devworkspace_name=empty-715 -n admin-devspaces -o name
pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db
          ▼ KubernetesCommandLineToolsExecutor.getContainerName - oc: Get container name.
oc get pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db -o jsonpath='{.spec.containers[*].name}' -n admin-devspaces
che-code-runtime-description che-gateway

    ✔ Create empty workspace (181343ms)
    Clone public repo without previous setup
          ▼ ContainerTerminal.executeCommand - oc:
oc exec -i pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db -n admin-devspaces -c che-code-runtime-description che-gateway -- sh -c "pwd"
/projects
          ▼ ContainerTerminal.executeCommand - oc:
oc exec -i pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db -n admin-devspaces -c che-code-runtime-description che-gateway -- sh -c "git clone https://github.com/crw-qe/web-nodejs-sample"
Cloning into 'web-nodejs-sample'...
      ✔ Check if public repo can be cloned (5687ms)
          ▼ Function.getProjectNameFromGitUrl - https://github.com/crw-qe/web-nodejs-sample
          ▼ Function.getProjectNameFromGitUrl - web-nodejs-sample
          ▼ ContainerTerminal.executeCommand - oc:
oc exec -i pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db -n admin-devspaces -c che-code-runtime-description che-gateway -- sh -c "ls "
web-nodejs-sample
      ✔ Check if project was created (2399ms)
          ▼ ContainerTerminal.executeCommand - oc:
oc exec -i pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db -n admin-devspaces -c che-code-runtime-description che-gateway -- sh -c "ls /projects/web-nodejs-sample"
LICENSE
README.md
app
devfile.yaml
package-lock.json
package.json
      ✔ Check if files were imported  (2011ms)
          ▼ ContainerTerminal.executeCommand - oc:
oc exec -i pod/workspacecfe20d63ed9d40e0-7d55756f94-nt6db -n admin-devspaces -c che-code-runtime-description che-gateway -- sh -c "rm -rf web-nodejs-sample"
          ▼ KubernetesCommandLineToolsExecutor.deleteDevWorkspace - oc: Delete 'empty-715' workspace
oc patch dw empty-715 -n admin-devspaces -p '{ "metadata": { "finalizers": null }}' --type merge || true
devworkspace.workspace.devfile.io/empty-715 patched
oc delete dw empty-715 -n admin-devspaces || true
devworkspace.workspace.devfile.io "empty-715" deleted
oc delete dwt che-code-empty-715 -n admin-devspaces || true
devworkspacetemplate.workspace.devfile.io "che-code-empty-715" deleted


  4 passing (3m)

            ‣ DriverHelper.wait (5000 milliseconds)

Process finished with exit code 0
```

File: tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    });
+
+    test('Create empty workspace', async function (): Promise<void> {
+        kubernetesCommandLineToolsExecutor.loginToOcp();
+        const devWorkspaceConfigurationYamlString: string = await devWorkspaceConfigurationHelper.getDevWorkspaceConfigurationYamlAsString(devfileContext);
+        const output: ShellString = kubernetesCommandLineToolsExecutor.applyAndWaitDevWorkspace(devWorkspaceConfigurationYamlString);
+        expect(output.stdout).contains('condition met');
+    });
+
+    suiteTeardown('Delete workspace', function (): void {
```
Comment (dmytro-ndp) [inline tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts:45]: Tear-down steps are usually put at the end of test script, for the sake of readiness.

Comment (dmytro-ndp) [inline tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts]: tear-down steps are usually placed at the end of test, for the sake of readiness

File: tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        kubernetesCommandLineToolsExecutor.deleteDevWorkspace();
+    });
+
+    suite('Clone public repo without previous setup', function (): void {
+        suiteTeardown('Delete cloned project', function (): void {
```
Comment (dmytro-ndp) [inline tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts]: tear-down steps are usually placed at the end of test, for the sake of readiness

File: tests/e2e/utils/DevWorkspaceConfigurationHelper.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { V1alpha2DevWorkspaceTemplate } from '@devfile/api';
 import YAML from 'yaml';
 import * as axios from 'axios';
 import { Logger } from './Logger';
+import { TestConstants } from '../constants/TestConstants';
+import { ShellExecutor } from './ShellExecutor';
 
 interface IContextParams {
```
Comment (dmytro-ndp) [inline tests/e2e/utils/DevWorkspaceConfigurationHelper.ts:15]: It's not clear what `IContextParams` is used for in the tests.
E.g., I can't find the usage of `injectDefaultComponent` and `defaultComponentImage` in API tests.

What do you think about adding a comment to the code with more details?

**Review by dmytro-ndp — APPROVED** [review]
Looks commonly good to merge, apart from a few non-critical notices.

File: tests/e2e/utils/DevWorkspaceConfigurationHelper.ts (hunk 3 above)
Comment (nallikaea) [reply tests/e2e/utils/DevWorkspaceConfigurationHelper.ts:15]: It used to see which exactly fields you can pass into `new DevWorkspaceConfigurationHelper(params);` as params:
image

Comment (nallikaea) [reply tests/e2e/utils/DevWorkspaceConfigurationHelper.ts:15]: Here is used `injectDefaultComponent` and `defaultComponentImage`, it described in DevWorkspace genetatortool read me
image

Comment (nallikaea) [reply tests/e2e/utils/DevWorkspaceConfigurationHelper.ts:15]: I will add link to the readme as comment

File: tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts (hunk 1 above)
Comment (nallikaea) [reply tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts]: done

File: tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts (hunk 2 above)
Comment (nallikaea) [reply tests/e2e/specs/api/EmptyWorkspaceAPI.spec.ts]: done

File: tests/e2e/utils/DevWorkspaceConfigurationHelper.ts (hunk 3 above)
Comment (nallikaea) [reply tests/e2e/utils/DevWorkspaceConfigurationHelper.ts:15]: done

**Pull request merged**